zulip/zulip/__init__.py

```diff
@@ -609,8 +609,8 @@ def call_endpoint(self, url: Optional[str] = None, method: str = "POST", request
             if v is not None:
                 marshalled_request[k] = v
         versioned_url = API_VERSTRING + (url if url is not None else "")
-        return self.do_api_query(marshalled_request, versioned_url, method=method,
-                                 longpolling=longpolling, files=files, timeout=timeout)
+        return json.dumps(self.do_api_query(marshalled_request, versioned_url, method=method,
```

Comment:
We definitely don't want to do JSON dumps here -- a program accessing this wants the Python dictionary.

What I think we want to do here is sorting in #663 and in all the tools in `examples/`, so the main PR to do here would be to adjust `examples/` appropriately.

Reply:
@timabbott Can we instead change this to 
```return json.loads(json.dumps(self.do_api_query(marshalled_request, versioned_url, method=method,
                                 longpolling=longpolling, files=files, timeout=timeout), sort_keys=True))```
since dictionaries are ordered since Python 3.7?

Reply:
@timabbott Should I instead go with the modifying `examples/` approach?

Reply:
@MSurfer20 I think we should go for changing `examples/`, but am not sure if it is worth solving the problem.

Reply:
I don't think this will help if someone is running zulip in an env where `python <= 3.6`.
The only way is updating `examples/` but I am not sure if it is worth the amount of work.

Reply:
I think after thinking about this more, I'm not sure there's anything we want to do client-side -- the extra code in the examples is unlikely to be worth it.